qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -12,53 +29,176 @@ class N9030B(VisaInstrument):
+            vals=Enum(*self._available_modes())
+        )
+
+        self.add_parameter(
+            name="measurement",
+            get_cmd=":CONFigure?",
+            set_cmd=":CONFigure:{}",
+            val_mapping={"Swept SA": "SANalyzer"}
+        )
+
+        self.add_parameter(
+            name="start",
+            get_cmd=":SENSe:FREQuency:STARt?",
+            set_cmd=self._set_start,
+            get_parser=float,
+            vals=Numbers(self._min_freq, self._max_freq - 10)
+        )
+
+        self.add_parameter(
+            name="stop",
+            get_cmd=":SENSe:FREQuency:STOP?",
+            set_cmd=self._set_stop,
+            get_parser=float,
+            vals=Numbers(self._min_freq + 10, self._max_freq)
+        )
+
+        self.add_parameter(
+            name="center",
+            get_cmd=":SENSe:FREQuency:CENTer?",
+            set_cmd=self._set_center,
+            get_parser=float,
+            vals=Numbers(self._min_freq + 5, self._max_freq - 5)
+        )
+
+        self.add_parameter(
+            name="span",
+            get_cmd=":SENSe:FREQuency:SPAN?",
+            set_cmd=self._set_span,
+            get_parser=float,
+            vals=Numbers(10, self._max_freq - self._min_freq),
+        )
+
+        self.add_parameter(
+            name="npts",
+            get_cmd=":SENSe:SWEep:POINts?",
+            set_cmd=self._set_npts,
+            get_parser=int,
+        )
+
+        self.add_parameter(
+            name="sweep_time",
+            label="Sweep time",
+            get_cmd=":SENSe:SWEep:TIME?",
+            get_parser=float,
+            unit="s",
+        )
+
+        self.add_parameter(
+            name="auto_sweep_time_enabled",
+            initial_value=False,
+            get_cmd=":SENSe:SWEep:TIME:AUTO?",
+            set_cmd=self._enable_auto_sweep_time,
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF")
+        )
+
+        self.add_parameter(
+            name="auto_sweep_type_enabled",
+            initial_value=False,
+            get_cmd=":SENSe:SWEep:TYPE:AUTO?",
+            set_cmd=self._enable_auto_sweep_type,
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF")
+        )
+
+        self.add_parameter(
+            name="sweep_type",
+            get_cmd=":SENSe:SWEep:TYPE?",
+            set_cmd=self._set_sweep_type,
+            val_mapping={
+                "fft": "FFT",
+                "sweep": "SWE",
+            }
+        )
+
+        self.add_parameter(
+            name='freq_axis',
+            label='Frequency',
+            unit='Hz',
+            vals=Arrays(shape=(self.npts,)),
+            parameter_class=FrequencyAxis
+        )
+
+        self.add_parameter(
+            name='trace',
+            vals=Arrays(shape=(self.npts,)),
+            setpoints=(self.freq_axis,),
+            parameter_class=Trace
+        )
+
+        self.add_function("reset", call_cmd="*RST")
```

Comment:
Please don't use the `.add_function` method. We're planning to remove that method, since it doesn't add any value (see also [the docs](https://qcodes.github.io/Qcodes/examples/writing_drivers/Creating-Instrument-Drivers.html#Functions).

A good old
```python
def reset(self) -> None:
    """
    Reset the instrument by sending the RST command
    """
    self.write("*RST")
```
will do the same and is easier to read and understand (and has a docstring).